### PR TITLE
Update verify.yml

### PR DESCRIPTION
**Pull request by volkflo**
I checked this morning that `apiman/on-wildfly:latest` was updated but It seems that the upload of the devportal never worked. I think the if-condition did not work the last time.

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
   Publish:
     needs: [Build, Test]
     runs-on: ubuntu-latest
-    if: github.event.pull_request.merged == true && github.repository_owner == 'Apiman'
+    if: github.ref == 'refs/heads/main' && github.repository_owner == 'Apiman'
```
Comment (bekihm) [inline .github/workflows/verify.yml:60]: And this works with this change?

Comment (volkflo) [reply .github/workflows/verify.yml:60]: I hope so, this is now the same code as in the apiman `verify.yml`

Comment (bekihm) [reply .github/workflows/verify.yml:60]: Ok 👍

**Review by bekihm — APPROVED** [review]

**Comment by volkflo** [comment]
image
Something is still wrong with the login, but at least the publish workflow stage was reached.

**Comment by volkflo** [comment]
Okay, It seems we have a little mismatch with our secretes:

apiman:
image
apiman-developer-portal:
image

@EricWittmann Has the DOCKERHUB_ACCESS_TOKEN the same value as the DOCKERHUB_PASSWORD or is this really a token and we need no username anymore and have to switch to a different login? :)

**Comment by volkflo** [comment]
@EricWittmann Seems like this has been forgotten. Can you please check my last comment with the secrets? :)

**Comment by EricWittmann** [comment]
Sorry, I missed that comment. :( :(

A docker hub access token works just like a password.  I'll just rename the secret and we should be in business.  Not sure what I was thinking.

**Comment by EricWittmann** [comment]
Done.  Let me know if it doesn't work.  The nice thing about access tokens (obviously) is that they can be scoped and revoked.

**Comment by volkflo** [comment]
Thanks, @EricWittmann! I also only remembered because of the other docker issues I closed before.
The good news: Docker login works!
The bad news:
image
I do not have any idea, why the runner is not able to find the dockerfile

**Comment by EricWittmann** [comment]
There's no Dockerfile in the root of the project.  The action does this:

image

But it looks like the Dockerfile is located in `apiman-developer-portal/docker/images/test`.

**Comment by EricWittmann** [comment]
Maybe something like this instead?

```yaml
steps:
      # Build and push new Docker images
      - name: Login to DockerHub Registry
        run: echo ${{ secrets.DOCKERHUB_PASSWORD }} | docker login -u ${{ secrets.DOCKERHUB_USERNAME }} --password-stdin
      - name: Building Docker Images
        run: cd docker && docker build . --file ./images/test/Dockerfile --tag apiman/developer-portal:latest
      - name: Push The Tagged Docker Images
        run: docker push apiman/developer-portal:latest
```

**Comment by volkflo** [comment]
@EricWittmann There is a Dockerfile in the project root: "https://github.com/apiman/apiman-developer-portal/blob/main/Dockerfile". So that confuses me a little bit.

**Comment by EricWittmann** [comment]
Oh geez - you are correct.  I have no idea why I didn't notice that.

**Comment by EricWittmann** [comment]
Maybe try this:

`docker build -t="apiman/developer-portal:latest" --rm .`

It should be the same thing, but I don't have any other ideas.

**Comment by bekihm** [comment]
@volkflo I think @EricWittmann is right you have to put the . After the the parameters:
docker build - - parameter .

**Comment by volkflo** [comment]
Yeah, both of you are correct. But there was another error: The checkout step for this stage was missing.
But now we have an image available on docker hub: https://hub.docker.com/r/apiman/developer-portal.

@EricWittmann Maybe you could update the readme on docker hub and link to the project or show us how to do this? Because the page is now empty :)

**Comment by EricWittmann** [comment]
I've copied the Readme content from the root of the github repo as the Readme in docker hub.  The only way to do that afaik is to edit the field manually in the docker hub UI.  To do that you need access privileges.  I think I can grant that if you send me your docker hub IDs. :)

**Pull request merged**